pkg/network/tcp_peer.go

```diff
@@ -174,7 +175,14 @@ func (p *TCPPeer) handleConn() {
 			} else if err != nil {
 				break
 			}
-			p.incoming <- msg
+			select {
+			case p.incoming <- msg:
+			case <-p.done:
```

Comment:
This needs to be the first in `select` then.

Reply:
Why?

Reply:
Sorry, I was thinking about something else.